cmd/placement/cmd/placement.go

```diff
@@ -134,18 +159,64 @@ func Run(ctx context.Context, options *placementoptions.Options) error {
 		kubeSharedInformerFactory.Core().V1().Namespaces(),
 		kcpSharedInformerFactory.Scheduling().V1alpha1().Locations(),
 		kcpSharedInformerFactory.Scheduling().V1alpha1().Placements(),
+		edgeSharedInformerFactory.Edge().V1alpha1().EdgePlacements(),
 	)
 	if err != nil {
 		logger.Error(err, "Failed to create controller")
 		return err
 	}
 
-	// run the kcp-scheduling-placement-controller
+	// run the edge-scheduler
 	kubeSharedInformerFactory.Start(ctx.Done())
 	kcpSharedInformerFactory.Start(ctx.Done())
+	edgeSharedInformerFactory.Start(ctx.Done())
 	kubeSharedInformerFactory.WaitForCacheSync(ctx.Done())
 	kcpSharedInformerFactory.WaitForCacheSync(ctx.Done())
+	edgeSharedInformerFactory.WaitForCacheSync(ctx.Done())
 	c.Start(ctx, 1)
 
 	return nil
 }
+
+func configForViewOfExport(ctx context.Context, providerConfig *rest.Config, exportName string) (*rest.Config, error) {
+	providerClient, err := kcpclientsetnoncluster.NewForConfig(providerConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating client for service provider workspace: %w", err)
+	}
+	apiExportClient := providerClient.ApisV1alpha1().APIExports()
+	logger := klog.FromContext(ctx)
+	var apiExport *apisv1alpha1.APIExport
+	for {
+		apiExport, err = apiExportClient.Get(ctx, exportName, metav1.GetOptions{})
+		if err != nil {
+			if k8sapierrors.IsNotFound(err) {
+				logger.Info("Pause because APIExport not found", "exportName", exportName)
+				time.Sleep(time.Second * 15)
```

Comment:
I think it is probably OK here. Because this waiting happens before we start the scheduler.

We need this APIExport as a prerequisite of the scheduler, because it provides us with the view of our custom resources.

Actually this waiting logic is incorporated from #147 , and I think it works well.

Reply:
Note that this pause is during the setup phase, it is not in a worker thread.

Reply:
Even if it is in the setup phase it should be tolerant of the APIExport not being available right away. How should we handle the case where the APIExport is added after the controller is started ? Should we let the controller crash so that the pod where is it running is restarted by Kube ?

Reply:
This retry loop *is* handling the case where the APIExport is added after the controller is started (along with other transient problems).

Kubernetes recognizes that a pod may take some startup time before it enters into regular service.  We could add a `/livez` handler to expose that.  In a later PR.